sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_common/_response_handlers.py

```diff
@@ -42,12 +42,22 @@ def _parse_response_schema_properties(response, format):
     properties_dict["format"] = SchemaFormat(format)
     return SchemaProperties(**properties_dict)
 
+def _get_format(content_type: str) -> SchemaFormat:
+    # pylint:disable=redefined-builtin
+    try:
+        format = content_type.split("serialization=")[1]
+        try:
+            format = SchemaFormat(format)
+        except ValueError:
+            format = SchemaFormat(format.capitalize())
+    except IndexError:
+        format = SchemaFormat.CUSTOM
```

Comment:
I am not sure how to feel about this detection logic for the custom format. On one hand, it is not precise, i.e. it doesn't check for `text/plain; charset=utf-8`. On the other hand, I don't think the service could return anything else in this case but who knows! Just bringing this up for discussion.

Reply:
I felt comfortable doing this since we generate from the swagger that specifies that these are the only content types that the service can return. + in Python, the convention is to use try/except rather than if/else.

I'll change this check if we find out that the service will return other content types/the swagger is updated.

Reply:
I agree that we should tailor our code to the swagger but my main concern is about the customer experience in the case the service didn't match the swagger because of a bug or something. JS autorest generates code that guards against unknown content types, see [[here]](https://github.com/Azure/azure-sdk-for-js/blob/e0b2d74424251abcdf7a27f894d18a10833aae77/sdk/automation/arm-automation/src/operations/dscConfigurationOperations.ts#L218-L244) for an example. I wonder if Python has a similar pattern and whether it is applicable here. /cc @lmazuel

Reply:
It looks like in JS, you're throwing an error. Is this error surfaced to the user? Or is it caught and handled some other way somewhere else in the code?

B/c I don't think we'd want to throw an error that's surfaced to the user. Since this is a service response, if they are returning an unknown content type, or a content type with a bug in it, then there's nothing the user can do to fix it anyway. They'd have to wait for the service fix.

Am I interpreting this correctly?

Reply:
The error is surfaced to the customer.

> Since this is a service response, if they are returning an unknown content type, or a content type with a bug in it, then there's nothing the user can do to fix it anyway. They'd have to wait for the service fix.

Yeah, but we should make them aware there is an issue on the service side instead of silently passing unexpected stuff to them.

Reply:
In the error message that you have [[here](https://github.com/Azure/azure-sdk-for-js/blob/e0b2d74424251abcdf7a27f894d18a10833aae77/sdk/automation/arm-automation/src/operations/dscConfigurationOperations.ts#L242)], it sounds like this is an error that is caused by the user/one that they can fix. It doesn't really seem to indicate that this is a service issue.

+ I don't know how likely it is that the service responses will not match the swagger. And if they don't, seems like either it would either be a bug or something that they'd notify us about + update the swagger. Thoughts @lmazuel?

Reply:
Actually, this is a poor example, the content type in that case is indeed chosen by the customer/convenience-layer before sending a request. Let me check what is the logic on the receiving side.

Reply:
I was mistaken about the autorest part, the generated JS code doesn't guard against unknown content types, the `produces` list in the swagger doesn't seem to be taken into consideration in the generated code. I wonder if that is intentional and why?

In my experience in the realm of LROs, the client needs to guard against unexpected situations and error accordingly to provide a good customer experience in the case of bugs on the service side. I wonder why the same approach is not followed in the case of unexpected content types.

Reply:
Ah, I didn't know that about LROs. I'm not sure why this is the case for generated code, but might be something to ask autorest folks?